### ISSUE TITLE
DocsResultRowValue does not include 'deleted' field

### ISSUE THREAD

**Issue by jnhagelberg** [IBM/cloudant-java-sdk#350]
**Describe the bug**
The DocsResultRowValue class (https://github.com/IBM/cloudant-java-sdk/blob/main/modules/cloudant/src/main/java/com/ibm/cloud/cloudant/v1/model/DocsResultRowValue.java) is missing the "deleted" field.  As a result we are unable to detect whether or not the fetched document(s) were deleted.

Here is an example of an _all_docs response that includes this field:
```
{
  "total_rows": 20,
  "offset": null,
  "rows": [
    {
      "id": "ec51759e521fccd79e60ed53a422db05",
      "key": "ec51759e521fccd79e60ed53a422db05",
      "value": {
        "rev": "2-eec205a9d413992850a6e32678485900",
        "deleted": true
      }
    }
  ]
}
```
Deleted documents are included if you explicitly specify the id of a deleted document as a key when querying the _all_docs view.

We are seeing this when using the postAllDocs api.

**Comment by tomblench** [comment]
@jnhagelberg you're right, this is a bug, and the `deleted` attribute is missing from the OpenAPI definition, from which the model class is generated.

Until we fix this and do a release you have a few options to work around the issue:
- Patch the DocsResultRowValue class locally by adding the `deleted` field
- If your documents are not too large, pass `includeDocs=true`, call `getDocument()` and check the `deleted` flag (whilst noting that this is a `Boolean` object which could be null (you can use `Optional.ofNullable(d.isDeleted()).orElse(Boolean.false)`)
- Use streaming I/O and perform your own JSON deserialisation

**Comment by jnhagelberg** [comment]
Thanks!

**Comment by ricellis** [comment]
This is available in the [0.4.0 release](https://github.com/IBM/cloudant-java-sdk/releases/tag/v0.4.0).
Please note as above the method returns a `Boolean` which may be `null` when the property is absent from the JSON.

**Comment by jnhagelberg** [comment]
Thanks!!
